src/riak_kv_wm_index.erl

```diff
@@ -316,7 +315,7 @@ validate_timeout(Str) ->
             {error, Str}
     end.
 
-validate_max(all) ->
+validate_max(undefined) ->
```

Comment:
slightly concerned about the mismatch LessThanZero below. Can be equal to zero.
But this has nothing to do with this refactoring for dialyzer

Reply:
I've changed the variable name for now.  Zero would make no sense